src/server/initServer.ts

```diff
@@ -267,13 +263,144 @@ const initServer = (
           );
         }
 
+        // Build MongoDB query based on filters
+        const query: any = {};
+
+        /* -------------- Date Filter ------------- */
+        const { startDate, endDate } = filters.dateFilterState;
+        const startTimestamp = new Date(startDate.year, startDate.month - 1, startDate.day).getTime();
+        const endTimestamp = new Date(endDate.year, endDate.month - 1, endDate.day + 1).getTime() - 1;
+
+        query.timestamp = {
+          $gte: startTimestamp,
+          $lte: endTimestamp,
+        };
+
+        /* ------------ Context Filter ------------ */
+        const { contextFilterState } = filters;
+        const contextConditions: any[] = [];
```

Comment:
Be more precise than "any"

Reply:
Could we go for a more straightforward query for contexts? Something like:

```ts
{
  context: { $in: selectedContexts },
  subcontext: { $in: selectedSubcontexts },
}
```

Contexts and sub contexts _must_ be unique, so we don't need to be super precise here.